src/System.Management.Automation/engine/ExperimentalFeature/EnableDisableExperimentalFeatureCommand.cs

```diff
@@ -112,26 +113,28 @@ public class ExperimentalFeatureNameCompleter : IArgumentCompleter
         /// <param name="commandAst">The command AST.</param>
         /// <param name="fakeBoundParameters">The fake bound parameters.</param>
         /// <returns>List of Completion Results.</returns>
-        public IEnumerable<CompletionResult> CompleteArgument(string commandName, string parameterName, string wordToComplete, CommandAst commandAst, IDictionary fakeBoundParameters)
+        public IEnumerable<CompletionResult> CompleteArgument(
+            string commandName,
+            string parameterName,
+            string wordToComplete,
+            CommandAst commandAst,
+            IDictionary fakeBoundParameters)
         {
-            if (fakeBoundParameters == null)
-            {
-                throw PSTraceSource.NewArgumentNullException(nameof(fakeBoundParameters));
-            }
-
-            var commandInfo = new CmdletInfo("Get-ExperimentalFeature", typeof(GetExperimentalFeatureCommand));
-            var ps = System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace)
-                .AddCommand(commandInfo)
-                .AddParameter("Name", wordToComplete + "*");
+            SortedSet<string> expirmentalFeatures = new(StringComparer.OrdinalIgnoreCase);
 
-            HashSet<string> names = new HashSet<string>();
-            var results = ps.Invoke<ExperimentalFeature>();
-            foreach (var result in results)
+            foreach (ExperimentalFeature feature in GetExperimentalFeatures())
             {
-                names.Add(result.Name);
+                expirmentalFeatures.Add(feature.Name);
             }
 
-            return names.Order().Select(static name => new CompletionResult(name, name, CompletionResultType.Text, name));
+            return CompletionCompleters.GetMatchingResults(wordToComplete, expirmentalFeatures);
+        }
+
+        private static Collection<ExperimentalFeature> GetExperimentalFeatures()
+        {
+            using var ps = System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace);
+            ps.AddCommand("Get-ExperimentalFeature");
+            return ps.Invoke<ExperimentalFeature>();
```

Comment:
If this is one of those cases for which we discussed new helper method, then it's clearly not worth it.

Reply:
Yeah don't think so since this needs no parameter added so not worth it.